### ISSUE TITLE
Chapter 5: can't reproduce "often end up with more than 95% accuracy in less than 10 epochs"

### ISSUE THREAD

**Issue by jeffhgs** [maxpumperla/deep_learning_and_the_game_of_go#17]
Chapter 5 reads:

> But it’s noteworthy to observe that you often end up with more than 95% accuracy in less than 10 epochs.

However, I don't share this experience.

I have captured some scenarios in a branch.

    https://github.com/jeffhgs/deep_learning_and_the_game_of_go/blob/test_nn_chapter_5/code/dlgo/nn/test_nn.py

You can run as:

    (cd code/dlgo/nn && python test_nn.py)

The program downsamples mnist train and test data, train, repeats, and then computes summary statistics.

The program uses the unittest framework for regression testing, and it also emits json documents between scenarios for offline analysis.

The program depends on the incomplete branch I made for #12.

The typical fitting accuracy I get for the full mnist dataset is around 57%.

For purposes of helping your reproduction, I am holding back all inessential changes, including fix for performance related #11.

I tried to run all the configurations listed in the test, but unfortunately my job just ran past the 4h timeout I assigned it, so I'll have to up it and try again.  I will attach the 4 instance hours worth of data I have.

This is my configuration:

    host: AWS t2.large
    OS: ubuntu
    Python: 3.6.8

**Comment by jeffhgs** [comment]
[test_nn_jeffhgs_2019-02-20_try1.txt](https://github.com/maxpumperla/deep_learning_and_the_game_of_go/files/2887637/test_nn_jeffhgs_2019-02-20_try1.txt)

**Comment by jeffhgs** [comment]
[test_nn_jeffhgs_2019-02-20_try1_cropped.txt](https://github.com/maxpumperla/deep_learning_and_the_game_of_go/files/2887647/test_nn_jeffhgs_2019-02-20_try1_cropped.txt)

**Comment by jeffhgs** [comment]
[test_nn_jeffhgs_2019-02-20_try1b.txt](https://github.com/maxpumperla/deep_learning_and_the_game_of_go/files/2887650/test_nn_jeffhgs_2019-02-20_try1b.txt)

**Comment by jeffhgs** [comment]
Data from a longer run.  Note this timed out also, so doesn't end cleanly.  But you can clearly see the fit measurements.  Each full data set run just under 2h for 10 epochs.

[test_nn_jeffhgs_2019-02-20_try3.txt](https://github.com/maxpumperla/deep_learning_and_the_game_of_go/files/2896496/test_nn_jeffhgs_2019-02-20_try3.txt)

**Comment by maxpumperla** [comment]
that's pretty strange. last time I evaluated this, I certainly ended up with 95% plus, most of the time. I mean, not that it really matters, the lesson is that the algo learns. But I get your frustration, sorry for the inconvenience.

My suspicion is that I used a different learning rate altogether for my experiments.

**Comment by RockyrLiu** [comment]
I met the same problem. What can I do to solve it?

**Comment by wbendick** [comment]
@maxpumperla could you please provide code that achieves the accuracy reported in the book in a reproducible way?
